src/odemis/driver/andorshrk.py

```diff
@@ -1494,6 +1494,10 @@ def _doSetGrating(self, g):
         self.SetGrating(g)
         # By default the Shamrock library keeps the same wavelength
 
+        if self.axes["grating"].choices[g] == "mirror":
```

Comment:
What do you mean with "positions impossible to reach back". 

So far I wrote:
        # With a mirror as grating, the SR193 always stays physically at wavelength 0,
        # but it doesn't report it back after changing the grating to a mirror.
        # So this can lead to positions impossible to reach back.
        # Also, calling GetCalibration() on a mirror raises an error.
        # Setting the wavelength to 0 ensures that getPixelToWavelength() never tries to 
        # call GetCalibration(), but simply returns an empty wavelength list.